4b/Grains/src/Coordinator.cs

```diff
@@ -25,11 +41,28 @@ public async Task<string> RunTests()
                 GrainFactory.GetGrain<ITest2>(8).HelloWorldTest()
             };
 
+            testRun.Times.Start = DateTime.Now.ToString(CultureInfo.InvariantCulture);
             await Task.WhenAll(tests);
+            testRun.Times.Finish = DateTime.Now.ToString(CultureInfo.InvariantCulture);
             
             stopwatch.Stop();
+            
+            var serializer = new XmlSerializer(typeof(TestRun));
+            var serializerNamespaces = new XmlSerializerNamespaces();
+            serializerNamespaces.Add(
+                prefix: "",
+                ns: "");
+            
+            var writer = new StringWriter();
+            serializer.Serialize(
+                writer,
+                testRun,
+                serializerNamespaces);
+
+            var document = XDocument.Parse(writer.ToString());
+            document.Descendants().Attributes().Where(a => a.IsNamespaceDeclaration).Remove();
 
-            return await Task.FromResult($"All done! Elapsed time: {stopwatch.ElapsedMilliseconds}ms.");
+            return await Task.FromResult($"-- Test Execution Details --- \n {document}");
```

Comment:
That's good, but all we need to return is the actual trx only, nothing else. We want to feed that directly back to the pipeline.